File: Patterns/Qt/Decorator/Pizza Decorator com plugins/PizzaDecorator/pdaplication/uicontroller.cpp
#include "uicontroller.h"
#include "mainwindow.h"

UiController::UiController():
    _window(new class MainWindow)
{

}

UiController::~UiController()
{

}

MainWindow *UiController::window()
{
    return _window;
}

bool UiController::initialize()
{
    _window->show();
    return true;
}

File: Redes/Jogo da velha/src/controladores/UiController.java
package controladores;

import interfaces.IUiController;
import jogo.TicTacToe;

public class UiController implements IUiController {
    private TicTacToe window;
    
    @Override
    public boolean initialize() {
        window = new TicTacToe();
        
        
        
        //adicionar código
        
        
        
        
        return true;
    }

}
